beetsplug/parentwork.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+# Copyright 2017, Dorian Soergel.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Gets work title, disambiguation, parent work and its disambiguation,
+composer, composer sort name and performers
+"""
+
+from __future__ import division, absolute_import, print_function
+
+from beets import ui
+from beets.plugins import BeetsPlugin
+
+import musicbrainzngs
+
+
+def work_father(mb_workid, work_date=None):
+    """ This function finds the id of the father work given its id"""
+    work_info = musicbrainzngs.get_work_by_id(mb_workid,
+                                              includes=["work-rels",
+                                                        "artist-rels"])
+    if 'artist-relation-list' in work_info['work'] and work_date is None:
+        for artist in work_info['work']['artist-relation-list']:
+            if artist['type'] == 'composer':
+                if 'end' in artist.keys():
+                    work_date = artist['end']
+
+    if 'work-relation-list' in work_info['work']:
+        for work_father in work_info['work']['work-relation-list']:
+            if work_father['type'] == 'parts' \
+                    and work_father.get('direction') == 'backward':
+                father_id = work_father['work']['id']
+                return father_id, work_date
+    return None, work_date
+
+
+def work_parent(mb_workid):
+    """This function finds the parentwork id of a work given its id. """
+    work_date = None
+    while True:
+        (new_mb_workid, work_date) = work_father(mb_workid, work_date)
+        if not new_mb_workid:
+            return mb_workid, work_date
+        mb_workid = new_mb_workid
+    return mb_workid, work_date
+
+
+def find_parentwork(mb_workid):
+    """This function gives the work relationships (dict) of a parentwork
+    given the id of the work"""
+    parent_id, work_date = work_parent(mb_workid)
+    work_info = musicbrainzngs.get_work_by_id(parent_id,
+                                              includes=["artist-rels"])
+    return work_info, work_date
+
+
+class ParentWorkPlugin(BeetsPlugin):
+    def __init__(self):
+        super(ParentWorkPlugin, self).__init__()
+
+        self.config.add({
+            'auto': False,
+            'force': False,
+        })
+
+        self._command = ui.Subcommand(
```

Comment:
It's probably more appropriate to declare the `ui.Subcommand` inside the `commands` method instead of making it a field. (It's only used there.)

Reply:
I have no idea how to do this and how it works, I just copy-pasted it from another plugin. Isn't there a difference between ```_command``` and ```command```?

Reply:
Nope, they're just different variable names. If you move this into the `commands()` function, you can just call it `command` instead of `self._command` (i.e., a local variable instead of a field).